Added apply-time-mutation to docs (#42175)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

### config.kubernetes.io/apply-time-mutation

Type: Annotation

> Example: `config.kubernetes.io/apply-time-mutation: "true"`

Reviewer: Are you sure it's value should be in boolean? 
Please read : https://github.com/kubernetes-sigs/cli-utils/blob/master/README.md#apply-time-mutation
Let's correct the values first

Author: Hi @Ritikaa96, I am getting confused with the example value for config.kubernetes.io/apply-time-mutation annotation is a multiline string in YAML format, specifying a list of substitutions, each represented by a dictionary (or map). Each substitution specifies the sourceRef, sourcePath, targetPath, and optionally the token. So how do I mention that out or do I simply leave blank example value.
```
metadata:
  annotations:
    config.kubernetes.io/apply-time-mutation: |
      - sourceRef:
          kind: Pod
          name: pod-b
        sourcePath: $.status.podIP
        targetPath: $.spec.containers[?(@.name=="nginx")].env[?(@.name=="SERVICE_HOST")].value
        token: ${pod-b-ip}
      - sourceRef:
          kind: Pod
          name: pod-b
        sourcePath: $.spec.containers[?(@.name=="nginx")].ports[?(@.name=="tcp")].containerPort
        targetPath: $.spec.containers[?(@.name=="nginx")].env[?(@.name=="SERVICE_PORT")].value
        token: ${pod-b-port}
```
In this example, the multiline string contains two substitutions, each represented by a dictionary (indicated by the - before each item). Each dictionary specifies the sourceRef, sourcePath, targetPath, and token properties, which are necessary for the Applier to perform the field value substitution during the apply process. The ${pod-b-ip} and ${pod-b-port} are placeholders that will be replaced with actual values based on the source object pod-b during the apply process.

> Example: `config.kubernetes.io/apply-time-mutation: "pods"`

Reviewer 2: Try this:
```suggestion
Example:
```yaml
    config.kubernetes.io/apply-time-mutation: |
      - sourceRef:
          kind: Pod
          name: pod-a
        sourcePath: $.status.podIP
        targetPath: $.spec.containers[?(@.name=="example")].env[?(@.name=="SERVICE_HOST")].value
        token: ${pod-a-ip}
```

I think that's a good start

Used on: All objects

> This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://github.com/kubernetes-sigs/cli-utils/blob/master/README.md#apply-time-mutation). The annotation may specify one or more substitutions. 

Reviewer 2: ```suggestion
This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://kpt.dev/reference/annotations/apply-time-mutation/). The annotation may specify one or more substitutions.
```

Each substitution includes a source object, a source field path, and a target field path, with an optional token.
The source and target field paths are specified using JSONPath.
If the token is specified, the token is replaced in the target field value string with the source field value. 

> If the token is not specified, the target field value is replaced with the source field value. This alternatively allows either templated interpretation or type preservation.

Reviewer 2: You can leave most of this detail out - people can click on the link to learn more.

> On the other hand, a value of `"false"` can be used to indicate that the object's configuration should not be subject to any automatic mutations during apply-time. This means that the object should retain its exact configuration as specified by the user or external tool.

Reviewer: ``` suggestion
This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://github.com/kubernetes-sigs/cli-utils/blob/master/README.md#apply-time-mutation). The annotation may specify one or more substitutions. 
Each substitution includes a source object, a source field path, and a target field path, with an optional token.
The source and target field paths are specified using JSONPath.
If the token is specified, the token is replaced in the target field value string with the source field value. 
If the token is not specified, the target field value is replaced with the source field value. This alternatively allows either templated interpretation or type preservation.

```

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

### config.kubernetes.io/apply-time-mutation

Type: Annotation

Example: `config.kubernetes.io/apply-time-mutation:`
```yaml
    # Set $CONFIG_MANAGED_BY based on an annotation of a related ConfigMap
    # Source annotation key is app.kubernetes.io/managed-by
    config.kubernetes.io/apply-time-mutation: |
      - sourceRef:
          kind: ConfigMap
          name: example-config
        sourcePath: $.metadata.annotations.app\.kubernetes\.io/managed-by
        targetPath: $.spec.containers[?(@.name=="example")].env[?(@.name=="CONFIG_MANAGED_BY")].value
```

Used on: All objects

This annotation is used on the target object to be modified to configure [Apply-Time Mutation](https://kpt.dev/reference/annotations/apply-time-mutation/). The annotation may specify one or more substitutions. 

> Each substitution includes a source object, a source field path, and a target field path, with an optional token.
> If the token is not specified, the target field value is replaced with the source field value. This alternatively allows either templated interpretation or type preservation.

Reviewer: As stated by @sftim, you can leave out most of the info in favor of the link provided.
```suggestion

```
